### ISSUE TITLE
DynaSim error with compiling

### ISSUE THREAD

**Issue by davestanley** [DynaSim/DynaSim#237]
I'm not sure what is causing the error here, but something has changed since the October 2 version of the code (commit 6dcad76c168b8ac9201525ee7c01fb2ea51435f0) to produce this bug. Seems to be an issue with the compiler.

I will look into debugging this when I have time, but if someone has a quick fix that would be even better.

```
kramer_IB
Currently Loaded Modulefiles:
  1) matlab/2017a     2) git/2.6.3        3) libcurl/7.48.0
PREPARING STUDY:
Creating study directory: /projectnb/crc-nak/stanley/model-dnsim-kramer_IB/kramer_IB/study_d171015_t191044_batch01
Creating data directory: /projectnb/crc-nak/stanley/model-dnsim-kramer_IB/kramer_IB/study_d171015_t191044_batch01/data
Creating plot directory: /projectnb/crc-nak/stanley/model-dnsim-kramer_IB/kramer_IB/study_d171015_t191044_batch01/plots
created temporary lock file for this process: /projectnb/crc-nak/stanley/model-dnsim-kramer_IB/kramer_IB/study_d171015_t191044_batch01/.lock_20171015191054948_0
saving study file: /projectnb/crc-nak/stanley/model-dnsim-kramer_IB/kramer_IB/study_d171015_t191044_batch01/studyinfo.mat
deleting temporary lock file for this process: /projectnb/crc-nak/stanley/model-dnsim-kramer_IB/kramer_IB/study_d171015_t191044_batch01/.lock_20171015191054948_0

PREPARING SOLVER:
Creating solver directory /projectnb/crc-nak/stanley/model-dnsim-kramer_IB/kramer_IB/study_d171015_t191044_batch01/solve
Changing directory to /projectnb/crc-nak/stanley/model-dnsim-kramer_IB/kramer_IB/study_d171015_t191044_batch01/solve
Saving params.mat
Creating solver file: /projectnb/crc-nak/stanley/model-dnsim-kramer_IB/kramer_IB/study_d171015_t191044_batch01/solve/solve_ode_study_d171015_t191044_batch01.m
Compiling file: /projectnb/crc-nak/stanley/model-dnsim-kramer_IB/kramer_IB/study_d171015_t191044_batch01/solve/solve_ode_study_d171015_t191044_batch01.m
            to: /projectnb/crc-nak/stanley/model-dnsim-kramer_IB/kramer_IB/study_d171015_t191044_batch01/solve/solve_ode_study_d171015_t191044_batch01_mex
??? Function 'license' is not supported for code generation. Consider adding coder.extrinsic('license') at the top of the function to bypass code
generation.

Error in ==> reportUI Line: 3 Column: 7
Code generation failed: View Error Report
Removing failed compiled solve file: /projectnb/crc-nak/stanley/model-dnsim-kramer_IB/kramer_IB/study_d171015_t191044_batch01/solve/solve_ode_study_d171015_t191044_batch01.m
Error: 
	 in codegen (line 0)
	 in makeMex (line 112)
	 in dsPrepareMEX (line 58)
	 in dsGetSolveFile (line 199)
	 in dsSimulate/tryFn (line 925)
	 in dsSimulate (line 765)
	 in include_kramer_IB_simulate (line 27)
	 in kramer_IB (line 788)
Changing directory to /projectnb/crc-nak/stanley/model-dnsim-kramer_IB/kramer_IB
Error using codegen

Error in dsPrepareMEX>makeMex (line 112)
  eval(['codegen -d codemex -config cfg ',file]);
Error in dsPrepareMEX (line 58)
  makeMex(mfileInput, options); % mex-file solver for solve file
Error in dsGetSolveFile (line 199)
  solve_file = dsPrepareMEX(solve_file, options);
Error in dsSimulate/tryFn (line 925)
            options.solve_file = dsGetSolveFile(model,studyinfo,options); % store name of solver file in options struct
Error in dsSimulate (line 765)
    tryFn(nargout)
Error in include_kramer_IB_simulate (line 27)
    [data,studyinfo]=dsSimulate(spec,'tspan',tspan,'dt',dt,'downsample_factor',dsfact,'solver',solver,'coder',0,...
Error in kramer_IB (line 788)
include_kramer_IB_simulate; 
>> 
```

**Comment by davestanley** [comment]
Code works fine without compile mode. This is a serial, noncluster job (e.g. parallel_flag = 0; cluster_flag = 0)

**Comment by kupiqu** [comment]
I'm taking a look

**Comment by kupiqu** [comment]
The problem is in rng_wrapper. It calls reportUI, which calls license, which is not supported by the coder. But that's not the only issue. If I use coder.extrinsic to avoid the issue, then the coder tries to compile octave functions and fails. Also, Octave fails because doesn't understand what coder.extrinsic is.

I'll try to think a way to workaround the issue, which at the moment doesn't seem so obvious.

**Comment by kupiqu** [comment]
I think best strategy is to change rng_wrapper by rng in solve_file when compile_flag = 1

I'll do it tonight

**Comment by kupiqu** [comment]
Should be fixed now (commit befce525c35146ed61d434433daa4057541f16f3)

Please reopen, otherwise.

**Comment by davestanley** [comment]
Seems to be working
